### PR TITLE
DataLinks: Fix context menu not showing in singlestat-ish visualisations

### PR DESCRIPTION
**Pull request by dprokop**
Fixes regression introduced by https://github.com/grafana/grafana/pull/19699

### CODE REVIEW

**Review by torkelo — APPROVED** [review]
LGTM!

File: packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx [hunk 1]
```diff
@@ -152,17 +153,35 @@ const getContextMenuStyles = stylesFactory((theme: GrafanaTheme) => {
...
+        bottom: window.innerHeight < rect.bottom + rect.height + OFFSET,
+      };
+
+      setPositionStyles({
+        position: 'fixed',
+        left: collisions.right ? x - rect.width - OFFSET : x - OFFSET,
+        top: collisions.bottom ? y - rect.height - OFFSET : y + OFFSET,
+      });
+    }
+  }, [menuRef]);
```
Comment (aocenas) [inline packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx]: I am not sure about this. `menuRef` should be stable and should not change, maybe this should be `menuRef.current`?

File: packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx [hunk 2]
```diff
@@ -152,17 +153,35 @@ const getContextMenuStyles = stylesFactory((theme: GrafanaTheme) => {
 export const ContextMenu: React.FC<ContextMenuProps> = React.memo(({ x, y, onClose, items, renderHeader }) => {
   const theme = useContext(ThemeContext);
   const menuRef = useRef(null);
+  const [positionStyles, setPositionStyles] = useState(null);
+
+  useEffect(() => {
+    if (menuRef) {
```
Comment (aocenas) [inline packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx]: `menuRef` should be non null always, should this be `menuRef.current`?

File: packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx [hunk 3]
```diff
@@ -152,17 +153,35 @@ const getContextMenuStyles = stylesFactory((theme: GrafanaTheme) => {
...
+      const rect = menuRef.current.getBoundingClientRect();
```
Comment (aocenas) [inline packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx]: If we are getting measurements shouldn't we use `useLayoutEffect`?

Comment (dprokop) [reply packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx]: yes

File: packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx (hunk 1 above)
Comment (dprokop) [reply packages/grafana-ui/src/components/ContextMenu/ContextMenu.tsx]: yeah, you r right

**Pull request merged**